e2e/composer-min-open-end-version/test.php

```diff
@@ -0,0 +1,6 @@
+<?php
+
+\PHPStan\Testing\assertType('int<80100, max>', PHP_VERSION_ID);
```

Comment:
This could be until 80499 which is the max version allowed in the config. It'd help us prevent more typos and be consistent.

Reply:
it now takes the max into account.

I have also added the min-version

Reply:
Do not change anything about the min version. It might seem inconsistent to you, but PHPStan can support analysing PHP 5 codebases without major issues and this would be too annoying.

Please also add E2E tests what would PHPStan think about the `PHP_` constants with:

* something like `^5.6` in composer.json. The result should be as precise as possible  -meaning 50600-50699 (there was no 5.7)
* something like `^7.0` in composer.json. The result should be as precise as possible  -meaning 70000-70499 (there was no 7.5)

Reply:
I want to limit the `min` key in phpstan.neon still to 70100 (because PHPStan doesn't offer any specific logic to lower versions), but I don't want to flag asking for PHP 5 as an error when there's nothing configured about a PHP version in phpstan.neon or composer.json. (Or it's configured in a way that allows for PHP 5, like `<=8.3` or `^5.6`)

Reply:
Hope it's understandable.

Reply:
Coming to thing of it, there's a 3rd E2E test proposal:

* No version in composer.json, no version in phpstan.neon - assert the `PHP_` constant values.